client/boot/common.js

```diff
@@ -114,7 +118,7 @@ const loggedOutMiddleware = currentUser => {
 const oauthTokenMiddleware = () => {
 	if ( config.isEnabled( 'oauth' ) ) {
 		// Forces OAuth users to the /login page if no token is present
-		page( '*', require( 'auth/controller' ).checkToken );
+		page( '*', oathController.checkToken );
```

Comment:
It’s interesting use case. It is bundled despite being behind the feature flag at the moment so it can be imported at the top ofvthe file

Reply:
Idea for the future: Create a small Webpack plugin (similar to `DefinePlugin`) that resolves `config.blah()` calls at compile time. Could save more than a few KB :)

Reply:
I had exactly the same idea at the first glance :) However it’s more complex as it seems. We keep it in this shape on purpose, because it allows us to create one build file that works with all non-development envs. We expose config data from the sever using JS variable instantiated inside the script tag. It was a trade-off that allowed us to speed up docker build and thus deploy time by a few minutes. Jake knows all the rationale behind it much better 😃

Reply:
To be clear. I’m not saying we shouldn’t investigate it. It can be quite cool. We just need to evaluate if we gain enough that would justify having a dedicated build for Calypso env equal production.

Reply:
Oh, that's right, I remember now that there were different builds before. Nevermind then, it would be a step backwards :)

Altough I agree that a separate `production` build would be worth it if we can make as many time-consuming tricks to optimize for bundle size as possible.

Reply:
We gain a lot by having all of our environments be based off the same docker image.  That said, I'm open to breaking with that if the tradeoffs are worth it!  They weren't in the past.

For this case, where we have code that is used in stages that aren't production, I think there is a more simple solution involving separating out that code into its own chunk. We do this right now with the devdocs chunk which is only actually used in the wpcalypso stage even though its accessible in all of them.